src/rules/orderedImportsRule.ts

```diff
@@ -127,64 +170,105 @@ export class Rule extends Lint.Rules.AbstractRule {
 type Transform = (x: string) => string;
 const TRANSFORMS = new Map<string, Transform>([
     ["any", () => ""],
-    ["case-insensitive", (x) => x.toLowerCase()],
+    ["case-insensitive", x => x.toLowerCase()],
     ["lowercase-first", flipCase],
-    ["lowercase-last", (x) => x],
-    ["full", (x) => x],
-    ["basename", (x) => {
-        if (!ts.isExternalModuleNameRelative(x)) {
-            return x;
-        }
+    ["lowercase-last", x => x],
+    ["full", x => x],
+    [
+        "basename",
+        x => {
+            if (!ts.isExternalModuleNameRelative(x)) {
+                return x;
+            }
 
-        const splitIndex = x.lastIndexOf("/");
-        if (splitIndex === -1) {
-            return x;
+            const splitIndex = x.lastIndexOf("/");
+            if (splitIndex === -1) {
+                return x;
+            }
+            return x.substr(splitIndex + 1);
         }
-        return x.substr(splitIndex + 1);
-    }],
+    ]
 ]);
 
 enum ImportType {
     LIBRARY_IMPORT = 1,
     PARENT_DIRECTORY_IMPORT = 2, // starts with "../"
-    CURRENT_DIRECTORY_IMPORT = 3, // starts with "./"
+    CURRENT_DIRECTORY_IMPORT = 3 // starts with "./"
 }
 
 interface Options {
     groupedImports: boolean;
     importSourcesOrderTransform: Transform;
-    moduleSourcePath: Transform;
+    moduleSourceTransform: Transform;
     namedImportsOrderTransform: Transform;
+    groups: GroupOption[];
+}
+
+interface GroupOption {
+    name: string;
+    match: RegExp | null;
+    order: number;
 }
 
 interface JsonOptions {
     "grouped-imports"?: boolean;
     "import-sources-order"?: string;
     "named-imports-order"?: string;
     "module-source-path"?: string;
+    groups?: JsonGroupOption[];
+}
+
+interface JsonGroupOption {
+    name?: string;
+    match: string | null;
```

Comment:
`null` is not documented above. is it valid in the options?

Reply:
i might prefer not supporting null and expecting a `.*` match-everything pattern. bit of an API question there.

Reply:
I could see that.  The reason I allowed match to null originally was an implementation detail that could probably be avoided now.  I will change it and see how it works out.

Reply:
It worked out well so I removed it.  I still use an defaultGroup object internally for unmatched values, but now I use match of `/.*/` there.